workers/blescanmulti.py

```diff
@@ -0,0 +1,51 @@
+import time
+from interruptingcow import timeout
+from bluepy.btle import Scanner, DefaultDelegate
+from mqtt import MqttMessage
+import mqtt
+from workers.base import BaseWorker
+from logger import _LOGGER
+
+REQUIREMENTS = ['bluepy']
+
+class ScanDelegate(DefaultDelegate):
+  def __init__(self):
+    DefaultDelegate.__init__(self)
+
+  def handleDiscovery(self, dev, isNewDev, isNewData):
+    if isNewDev:
+      _LOGGER.debug("Discovered new device: %s" % dev.addr)
+
+class BlescanmultiWorker(BaseWorker):
+  def searchmac(self, devices, mac):
+    present = False
+    for dev in devices:
+      if dev.addr == mac.lower():
+         present = True
+
+    return present
+
+  def getrssi(self, devices, mac):
+    rssivalue = -999
+    for dev in devices:
+      if dev.addr == mac.lower():
+         rssivalue = dev.rssi
+
+    return rssivalue
+
+  def status_update(self):
+    scanner = Scanner().withDelegate(ScanDelegate())
+    devices = scanner.scan(10.0)
+    ret = []
+
+    for name, mac in self.devices.items():
+      try:
+        if self.searchmac(devices, mac):
+          ret.append(MqttMessage(topic=self.format_topic('presence/'+name+'/rssi'), payload=str(self.getrssi(devices,mac))))
+          ret.append(MqttMessage(topic=self.format_topic('presence/'+name), payload="1"))
+        else:
+          ret.append(MqttMessage(topic=self.format_topic('presence/'+name), payload="0"))
+      except RuntimeError:
```

Comment:
Why catching RuntimeError here ?

Reply:
What error is being thrown there any why ? Im not sure about Python error but RuntimeErrors seems very general. I also dont see where any exception came from in that code ?

Reply:
OK. It was just an attempt to capture any errors. In your opinion it would be better to leave only the ret.append?

Reply:
Gateway on the top level catches all exceptions and prints them. So its possible to see whats wrong if anything. Catching it here and not printing hides any possible errors and make it harder to debug setup.